lib/filter_stream.nit

```diff
@@ -79,12 +79,14 @@ class StreamCat
 		end
 	end
 
-	init with_streams(streams: Array[Reader])
+	init with_streams(d: nullable Decoder, streams: Array[Reader])
 	do
+		decoder = if d == null then new UTF8Decoder else d
 		_streams = streams.iterator
 	end
 
-	init(streams: Reader ...) is old_style_init do
+	init(d: nullable Decoder, streams: Reader ...) is old_style_init do
```

Comment:
I don't like it. This complexify the constructor where, for most use, we don't even want to know about that implementation detail.

Some suggestions:

1. do not use one letter parameters for standard library APIs, so I would go for `decoder`

2. Since it's more that optional (you use the default one if no one is passed) why don't you use a default constructor that takes no `decoder` and use the UTF-8 one and a named constructor that expects a decoder?

3. As a last resort you can use the optional arguments.